server/Episememe.Application/Filtering/BaseFiltering/PrivateMediaFilter.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Episememe.Domain.Entities;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Episememe.Application.Filtering.BaseFiltering
+{
+    public class PrivateMediaFilter : IFilter<MediaInstance>
+    {
+        private readonly string? _userId;
+
+        public PrivateMediaFilter(string? userId)
+        {
+            _userId = userId;
+        }
+
+        public ReadOnlyCollection<MediaInstance> Filter(ReadOnlyCollection<MediaInstance> instances)
+        {
+            return ConsiderPrivate(instances);
+        }
+
+        private ReadOnlyCollection<MediaInstance> ConsiderPrivate(ReadOnlyCollection<MediaInstance> mediaInstances)
+        {
+            var filteredMedia = mediaInstances;
+            if (!string.IsNullOrEmpty(_userId))
+            {
+                filteredMedia = mediaInstances.Where(mi => !mi.IsPrivate
+                                                           || (mi.IsPrivate && mi.AuthorId == _userId))
```

Comment:
Can be simplified a bit. It is equivalent to `!mi.IsPrivate || mi.AuthorId == _userId`